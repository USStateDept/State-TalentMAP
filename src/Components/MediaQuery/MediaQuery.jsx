import React from 'react';
import PropTypes from 'prop-types';
import MediaQuery from 'react-responsive';
import breakpoints from '../../sass/sass-vars/variables';

const MediaQueryWrapper = ({ breakpoint, widthType, children }) => {
  const mediaProps = {};

  // set the right prop based on the widthType
  if (widthType === 'max') { mediaProps.maxWidth = breakpoints[`${breakpoint}Num`]; }
  if (widthType === 'min') { mediaProps.minWidth = breakpoints[`${breakpoint}Num`]; }
  return (
    <MediaQuery {...mediaProps}>
      {children}
    </MediaQuery>
  );
};

MediaQueryWrapper.propTypes = {
  breakpoint: PropTypes.oneOf(['screenSmMin', 'screenXsMax', 'screenMdMin', 'screenSmMax', 'screenLgMin', 'screenMdMax']).isRequired,
  widthType: PropTypes.oneOf(['max', 'min']).isRequired,
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
};

export default MediaQueryWrapper;
