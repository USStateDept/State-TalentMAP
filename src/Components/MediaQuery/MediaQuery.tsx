import MediaQuery from 'react-responsive';
// @ts-ignore
import breakpoints from '../../sass/sass-vars/variables';

type Props = {
  breakpoint:'screenSmMin' | 'screenXsMax' | 'screenMdMin' | 'screenSmMax' | 'screenLgMin' | 'screenMdMax' | 'screenXlgMin' | 'screenLgMax';
  widthType: 'max' | 'min';
  children: React.ReactNode | Function; // eslint-disable-line @typescript-eslint/ban-types
};

const MediaQueryWrapper: React.FC<Props> = props => {
  const mediaProps = {
    maxWidth: undefined,
    minWidth: undefined,
  };

  const { breakpoint, widthType, children } = props;

  // set the right prop based on the widthType
  if (widthType === 'max') { mediaProps.maxWidth = breakpoints[`${breakpoint}Num`]; }
  if (widthType === 'min') { mediaProps.minWidth = breakpoints[`${breakpoint}Num`]; }
  return (
    <MediaQuery {...mediaProps}>
      {children}
    </MediaQuery>
  );
};

export default MediaQueryWrapper;
