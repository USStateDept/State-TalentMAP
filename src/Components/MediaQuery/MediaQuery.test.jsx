import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MediaQuery from './MediaQuery';

describe('MediaQueryComponent', () => {
  it('is defined when widthType is "max"', () => {
    const wrapper = shallow(
      <MediaQuery
        breakpoint="screenSmMin"
        widthType="max"
      >
        <div />
      </MediaQuery>,
    );
    expect(wrapper).toBeDefined();
  });

  it('is defined when widthType is "min"', () => {
    const wrapper = shallow(
      <MediaQuery
        breakpoint="screenSmMax"
        widthType="min"
      >
        <div />
      </MediaQuery>,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <MediaQuery
        breakpoint="screenSmMin"
        widthType="max"
      >
        <div />
      </MediaQuery>,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
