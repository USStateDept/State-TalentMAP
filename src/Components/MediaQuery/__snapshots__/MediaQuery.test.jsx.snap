// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MediaQueryComponent matches snapshot 1`] = `
<MediaQuery
  maxWidth={600}
>
  <div />
</MediaQuery>
`;
